Czaple/Models/IntroProvider.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  IntroProvider.swift
+//  Czaple
+//
+//  Created by Filip Trzaska on 12/02/2020.
+//  Copyright © 2020 Filip Trzaska. All rights reserved.
+//
+
+import UIKit
+
+protocol IntroProviding {
+    func showIntro(window: UIWindow, first: Bool)
+}
+
+final class IntroProvider: IntroProviding {
+    private let attractionProvider: AttractionProviderProtocol
+    private let boardAlert: BoardAlerting
+    
+    init(attractionProvider: AttractionProviderProtocol, boardAlert: BoardAlerting) {
+        self.attractionProvider = attractionProvider
+        self.boardAlert = boardAlert
+    }
+    
+    func showIntro(window: UIWindow, first: Bool) {
+        if first == true {
```

Comment:
`if first {` would work also. 
Generally I'm fan of returning values rather than changing them. you could simple return proper viewController like I mentioned earlier  :)